include/vapor/UnstructuredGrid3D.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include <ostream>
+#include <vector>
+#include <memory>
+#include <vapor/common.h>
+#include <vapor/UnstructuredGrid2D.h>
+#include <vapor/QuadTreeRectangle.hpp>
+
+
+#ifdef WIN32
+    #pragma warning(disable : 4661 4251)    // needed for template class
+#endif
+
+namespace VAPoR {
+
+//! \class UnstructuredGrid3D
+//! \brief class for a Layered unstructured grid.
+//!
+//
+class VDF_API UnstructuredGrid3D : public UnstructuredGrid {
+public:
+    //! Construct a unstructured grid sampling Layered scalar function
+    //!
+    //
+    UnstructuredGrid3D(const std::vector<size_t> &vertexDims, const std::vector<size_t> &faceDims, const std::vector<size_t> &edgeDims, const std::vector<size_t> &bs, const std::vector<float *> &blks,
+                       const int *vertexOnFace, const int *faceOnVertex, const int *faceOnFace,
+                       Location location,    // node,face, edge
+                       size_t maxVertexPerFace, size_t maxFacePerVertex, long nodeOffset, long cellOffset, const UnstructuredGridCoordless &xug, const UnstructuredGridCoordless &yug,
+                       const UnstructuredGridCoordless &zug);
+
+    UnstructuredGrid3D() = default;
+    virtual ~UnstructuredGrid3D() = default;
+
+    // std::shared_ptr<const QuadTreeRectangle<float, size_t> >GetQuadTreeRectangle() const {
```

Comment:
do you want to delete this unused function ?

Reply:
Done.